stablehlo/dialect/TypeInference.cpp

```diff
@@ -1818,56 +1818,57 @@ LogicalResult inferAfterAllOp(HloDialectInterface* dialect,
 }
 
 LogicalResult inferAllToAllOp(
-    std::optional<Location> location, Value operand, int64_t splitDimension,
-    int64_t concatDimension, int64_t splitCount,
+    std::optional<Location> location, ValueRange operands,
+    int64_t splitDimension, int64_t concatDimension, int64_t splitCount,
     DenseIntElementsAttr replicaGroups,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
   // all_to_all_c5, all_to_all_c7, all_to_all_i5
   if (failed(verifyReplicaGroups(location, replicaGroups,
                                  /*allGroupsMustHaveSameSize=*/true,
                                  /*useGlobalDeviceIds=*/false, splitCount)))
     return failure();
+  for (auto operand : operands) {
```

Comment:
I'm not sure whether `ValueRange` gives back a reference or a copy (my guess is a copy?), but let's make the intent explicit.

Reply:
changed `auto` -> `Value`, thanks